integration_tests/auction_test.go

```diff
@@ -0,0 +1,239 @@
+		s.Require().NoError(err)
+
+		proposal := types.SetTokenPricesProposal{
+			Title:       "initial token price submission",
+			Description: "our first token prices",
+			TokenPrices: []*types.ProposedTokenPrice{
+				{
+					Denom:    "gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af",
+					UsdPrice: sdk.MustNewDecFromStr("1.0"),
+				},
+				{
+					Denom:    "gravity0x5a98fcbea516cf06857215779fd812ca3bef1b32",
+					UsdPrice: sdk.MustNewDecFromStr("0.25"),
+				},
+				{
+					Denom:    types.UsommDenom,
+					UsdPrice: sdk.MustNewDecFromStr("0.5"),
+				},
+			},
+		}
+
+		proposalMsg, err := govtypes.NewMsgSubmitProposal(
+			&proposal,
+			sdk.Coins{
+				{
+					Denom:  testDenom,
+					Amount: stakeAmount.Quo(sdk.NewInt(2)),
+				},
+			},
+			orch.keyInfo.GetAddress(),
+		)
+		s.Require().NoError(err, "unable to create governance proposal")
+
+		s.T().Log("submit proposal")
+		submitProposalResponse, err := s.chain.sendMsgs(*orchClientCtx, proposalMsg)
+		s.Require().NoError(err)
+		s.Require().Zero(submitProposalResponse.Code, "raw log: %s", submitProposalResponse.RawLog)
+
+		s.T().Log("check proposal was submitted correctly")
+		govQueryClient := govtypes.NewQueryClient(orchClientCtx)
+		proposalsQueryResponse, err := govQueryClient.Proposals(context.Background(), &govtypes.QueryProposalsRequest{})
+		s.Require().NoError(err)
+		s.Require().NotEmpty(proposalsQueryResponse.Proposals)
+		s.Require().Equal(uint64(1), proposalsQueryResponse.Proposals[0].ProposalId, "not proposal id 1")
+		s.Require().Equal(govtypes.StatusVotingPeriod, proposalsQueryResponse.Proposals[0].Status, "proposal not in voting period")
+
+		s.T().Log("vote for proposal")
+		for _, val := range s.chain.validators {
+			kr, err := val.keyring()
+			s.Require().NoError(err)
+			localClientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kr, "val", val.keyInfo.GetAddress())
+			s.Require().NoError(err)
+
+			voteMsg := govtypes.NewMsgVote(val.keyInfo.GetAddress(), 1, govtypes.OptionYes)
+			voteResponse, err := s.chain.sendMsgs(*localClientCtx, voteMsg)
+			s.Require().NoError(err)
+			s.Require().Zero(voteResponse.Code, "vote error: %s", voteResponse.RawLog)
+		}
+
+		s.T().Log("wait for proposal to be approved")
+		s.Require().Eventuallyf(func() bool {
+			proposalQueryResponse, err := govQueryClient.Proposal(context.Background(), &govtypes.QueryProposalRequest{ProposalId: 1})
+			s.Require().NoError(err)
+			return govtypes.StatusPassed == proposalQueryResponse.Proposal.Status
+		}, time.Second*30, time.Second*5, "proposal was never accepted")
+
+		// Verify auction created for testing exists
+		auctionQuery := types.QueryActiveAuctionRequest{
+			AuctionId: uint32(1),
+		}
+		auctionResponse, err := auctionQueryClient.QueryActiveAuction(context.Background(), &auctionQuery)
+		s.Require().NoError(err)
+		s.Require().Equal(auctionResponse.Auction.Id, uint32(1))
```

Comment:
I'd have a similar point here about IDs as I mentioned earlier. In essence we're hardcoding these values in the setup_test.go so I think it kinda tracks to keeping them hardcoded here. That being said I'm open to doing otherwise